src/java.base/share/classes/java/io/DataInputStream.java

```diff
@@ -573,7 +573,7 @@ public static final String readUTF(DataInput in) throws IOException {
         byte[] bytearr = null;
         char[] chararr = null;
         if (in instanceof DataInputStream dis) {
-            if (dis.bytearr.length < utflen){
+            if (dis.bytearr == null || dis.bytearr.length < utflen){
```

Comment:
Add a space between ')' and '{'?

```suggestion
            if (dis.bytearr == null || dis.bytearr.length < utflen) {
```